week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,7 +1,32 @@
 // Implement the missing functionality according to the task description.
 // Hint: Remember which function you need to call when it's time to alarm.
 
-function setAlarm() { }
+let timer;
+
+function setAlarm() {
+  clearInterval(timer);
+  let inputVal = document.getElementById("alarmSet").value;
+  timer = setInterval(alarmCount, 1000);
+  function alarmCount() {
+    if (inputVal >= 0) {
+      document.getElementById(
+        "timeRemaining"
+      ).innerText = `Time Remaining: ${sec2time(inputVal)} `;
+      inputVal -= 1;
+      if (inputVal < 0) {
+        playAlarm();
+      }
+    }
+  }
```

Comment:
the code is really good
the only issue is readability 
no need to declare a function inside another function 
DOM elements declaration can be outside 
names of functions, sec2time ?

Reply:
- no need to declare a function inside another function
- DOM elements declaration can be outside
Tried but could not do these.